xla/pjrt/c/pjrt_c_api_gpu_internal.cc

```diff
@@ -167,12 +167,100 @@ PJRT_Error* PJRT_ExecuteContext_Create(PJRT_ExecuteContext_Create_Args* args) {
   return nullptr;
 }
 
+namespace {
+
+absl::StatusOr<std::unique_ptr<xla::StreamExecutorGpuTopologyDescription>>
+TryParseTopology(
+    xla::PjRtPlatformId platform_id, std::string platform_name,
+    absl::flat_hash_map<std::string, xla::PjRtValueType>& options) {
+  auto topology_it = options.find("topology");
```

Comment:
Who puts it there in the first place? The test just manually adds it to `options`, maybe we need a more e2e-like test?

Reply:
Currently, the options come from the kwargs of the [`jax.experimental.topologies.get_topology_desc`](https://github.com/google/jax/blob/baf9cc70bd4c9ad544615e54064fc209e1263fb9/jax/experimental/topologies.py#L49) call. See the [example in the gist](https://gist.github.com/jaro-sevcik/3495718bb04c6096c0f998fc29220c2b#file-aot-save-py-L54). So the idea is that the user will put it there.

By E2E example, do you mean creating the topology, then creating [compile-only-client](https://github.com/openxla/xla/blob/fe141265482a26d25318a5720d49963f9b45113c/xla/python/py_compile_only_client.cc#L352) and compiling with that client? Do we have a GPU-only XLA python test suite where I could put this?

Note that the compile-only part is already [somewhat tested in JAX](https://github.com/google/jax/blob/baf9cc70bd4c9ad544615e54064fc209e1263fb9/tests/aot_test.py#L103), this PR only makes it so that we do not require GPU device to run. (Unfortunately, I believe that the unit test does not really verify we can create the topology without a device.)